### ISSUE TITLE
ServiceBusAdministrationClient 400 error when adding subscription with forwardTo

### ISSUE THREAD

**Issue by sjw1066** [Azure/azure-sdk-for-java#24235]
**Describe the bug**
This bug appears to be very similar to [#8350](https://github.com/Azure/azure-sdk-for-java/issues/8350); in this case the `ServiceBusAdministrationClient` from `com.azure:azure-messaging-servicebus` will not successfully create a subscription with a forwardTo parameter, failing with a 400 - `BR0012 The specified value is in an invalid format.`

***Exception or Stack Trace***
```
Exception in thread "main" com.azure.core.exception.HttpResponseException: BR0012The specified value is in an invalid format.
Parameter name: ForwardTo. TrackingId:c82e1491-c2df-4a6f-b34f-0e5ca2fe1969_G11, SystemTracker:sb-ge-encompass-nprd:Topic:foo-poc-topic, Timestamp:2021-09-21T19:45:52
	at com.azure.messaging.servicebus.administration.ServiceBusAdministrationAsyncClient.mapException(ServiceBusAdministrationAsyncClient.java:2596)
	at reactor.core.publisher.Mono.lambda$onErrorMap$31(Mono.java:3488)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onError(FluxOnErrorResume.java:94)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onError(MonoFlatMap.java:172)
	at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onError(FluxContextWrite.java:121)
	at reactor.core.publisher.FluxDoOnEach$DoOnEachSubscriber.onError(FluxDoOnEach.java:195)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192)
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onError(Operators.java:2062)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.secondError(MonoFlatMap.java:192)
	at reactor.core.publisher.MonoFlatMap$FlatMapInner.onError(MonoFlatMap.java:259)
	at reactor.core.publisher.Operators$MultiSubscriptionSubscriber.onError(Operators.java:2062)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:142)
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1815)
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.signalCached(MonoCacheTime.java:337)
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.onNext(MonoCacheTime.java:354)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2397)
	at reactor.core.publisher.MonoCacheTime$CoordinatorSubscriber.onSubscribe(MonoCacheTime.java:293)
	at reactor.core.publisher.FluxFlatMap.trySubscribeScalarMap(FluxFlatMap.java:192)
	at reactor.core.publisher.MonoFlatMap.subscribeOrReturn(MonoFlatMap.java:53)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
	at reactor.core.publisher.MonoCacheTime.subscribeOrReturn(MonoCacheTime.java:143)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:57)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157)
	at reactor.core.publisher.Operators$ScalarSubscription.request(Operators.java:2397)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onSubscribe(MonoFlatMap.java:110)
	at reactor.core.publisher.MonoJust.subscribe(MonoJust.java:54)
	at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:157)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:120)
	at reactor.core.publisher.FluxHide$SuppressFuseableSubscriber.onNext(FluxHide.java:136)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1815)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:151)
	at reactor.core.publisher.SerializedSubscriber.onNext(SerializedSubscriber.java:99)
	at reactor.core.publisher.FluxRetryWhen$RetryWhenMainSubscriber.onNext(FluxRetryWhen.java:173)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
	at reactor.core.publisher.Operators$MonoInnerProducerBase.complete(Operators.java:2659)
	at reactor.core.publisher.MonoSingle$SingleSubscriber.onComplete(MonoSingle.java:180)
	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onComplete(MonoFlatMapMany.java:260)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:150)
	at reactor.core.publisher.FluxMapFuseable$MapFuseableSubscriber.onComplete(FluxMapFuseable.java:150)
	at reactor.core.publisher.Operators$MonoSubscriber.complete(Operators.java:1816)
	at reactor.core.publisher.MonoCollect$CollectSubscriber.onComplete(MonoCollect.java:159)
	at reactor.core.publisher.FluxDoFinally$DoFinallySubscriber.onComplete(FluxDoFinally.java:145)
	at reactor.core.publisher.FluxHandle$HandleSubscriber.onComplete(FluxHandle.java:212)
	at reactor.core.publisher.FluxMap$MapConditionalSubscriber.onComplete(FluxMap.java:269)
	at reactor.netty.channel.FluxReceive.onInboundComplete(FluxReceive.java:401)
	at reactor.netty.channel.ChannelOperations.onInboundComplete(ChannelOperations.java:416)
	at reactor.netty.channel.ChannelOperations.terminate(ChannelOperations.java:470)
	at reactor.netty.http.client.HttpClientOperations.onInboundNext(HttpClientOperations.java:685)
	at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:94)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1504)
	at io.netty.handler.ssl.SslHandler.decodeNonJdkCompatible(SslHandler.java:1265)
	at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1302)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:508)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:447)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.proxy.ProxyHandler.channelRead(ProxyHandler.java:253)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:253)
	at io.netty.handler.proxy.HttpProxyHandler$HttpClientCodecWrapper.channelRead(HttpProxyHandler.java:272)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99)
		at reactor.core.publisher.Mono.block(Mono.java:1703)
		at com.azure.messaging.servicebus.administration.ServiceBusAdministrationClient.createSubscription(ServiceBusAdministrationClient.java:236)
		at org.example.Main.main(Main.java:38)
Caused by: com.azure.messaging.servicebus.implementation.models.ServiceBusManagementErrorException: Status code 400, "<Error><Code>400</Code><Detail>BR0012The specified value is in an invalid format.&#xD;
Parameter name: ForwardTo. TrackingId:c82e1491-c2df-4a6f-b34f-0e5ca2fe1969_G11, SystemTracker:sb-ge-encompass-nprd:Topic:foo-poc-topic, Timestamp:2021-09-21T19:45:52</Detail></Error>"
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.azure.core.http.rest.RestProxy.instantiateUnexpectedException(RestProxy.java:343)
	at com.azure.core.http.rest.RestProxy.lambda$ensureExpectedStatus$5(RestProxy.java:382)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:125)
	... 83 more
```

**To Reproduce**
1- Attempt to create a new subscription on an existing topic that auto-forwards to another destination.

***Code Snippet***
```java
package org.example;


import com.azure.messaging.servicebus.administration.ServiceBusAdministrationClient;
import com.azure.messaging.servicebus.administration.ServiceBusAdministrationClientBuilder;
import com.azure.messaging.servicebus.administration.models.CreateQueueOptions;
import com.azure.messaging.servicebus.administration.models.CreateSubscriptionOptions;
import com.azure.messaging.servicebus.administration.models.CreateTopicOptions;

public class Main {

  private static final String CXN_STRING = "some-string-here";

  public static void main(String[] args) throws Exception {
    ServiceBusAdministrationClient adminClient = new ServiceBusAdministrationClientBuilder().connectionString(CXN_STRING).buildClient();

    String topicName = "foo-poc-topic";
    String queueName = "foo-poc-queue";

    String subName = "foo-fw-subscription";


    // create topic first
    CreateTopicOptions topicOpts = new CreateTopicOptions()
      .setBatchedOperationsEnabled(true)
      .setOrderingSupported(true);
    adminClient.createTopic(topicName, topicOpts);

    // now queue
    CreateQueueOptions queueOpts = new CreateQueueOptions()
      .setBatchedOperationsEnabled(true);
    adminClient.createQueue(queueName, queueOpts);

    // now try to create subscription on the topic that forwards to our queue
    CreateSubscriptionOptions subscriptionOpts = new CreateSubscriptionOptions()
      .setBatchedOperationsEnabled(true)
      .setForwardTo(queueName);
    adminClient.createSubscription(topicName, subName, subscriptionOpts);

  }
}
```

**Expected behavior**
I would expect the creation of a new subscription (named `foo-fw-subscription`) on the topic `foo-poc-topic` that auto-forwards messages to the queue `foo-poc-queue`.

**Setup (please complete the following information):**
 - Windows 10
 - IDE: IntelliJ 2021.2 Ultimate
 - Library/Libraries: com.azure:azure-messaging-servicebus:7.4.1
 - Java version: 11
 - App Server/Environment: N/A
 - Frameworks: Spring Boot

**Additional context**
Digging around, I found [#8558](https://github.com/Azure/azure-sdk-for-java/pull/8558) - I tried looking at the changes and attenpted to 'normalize' the forwardTo param:
```java
static String normalizeForwardToAddress(String forwardTo, URI baseAddress) {
    try {
      new URL(forwardTo);
      return forwardTo;
    } catch (MalformedURLException e) {
      return baseAddress.resolve(forwardTo).toString();
    }
  }

ServiceBusAdministrationClient adminClient = new ServiceBusAdministrationClientBuilder().connectionString(CXN_STRING).buildClient();
    ConnectionStringProperties props = new ConnectionStringProperties(CXN_STRING);
    URI baseEndpoint = props.getEndpoint();
...
...
// now try to create subscription on the topic that forwards to our queue
    CreateSubscriptionOptions subscriptionOpts = new CreateSubscriptionOptions()
      .setBatchedOperationsEnabled(true)
      .setForwardTo(normalizeForwardToAddress(queueName, baseEndpoint));
    adminClient.createSubscription(topicName, subName, subscriptionOpts);
```

This resulted in a change from 400 to 401.
```
Caused by: com.azure.messaging.servicebus.implementation.models.ServiceBusManagementErrorException: Status code 401, "<Error><Code>401</Code><Detail>MissingToken: The authorization header was not found. To know more visit https://aka.ms/sbResourceMgrExceptions. . TrackingId:fefec298-1dd6-4a22-8c1d-c8d1f4dc7780_G11, SystemTracker:sb-ge-encompass-nprd:Topic:foo-poc-topic, Timestamp:2021-09-21T20:01:28</Detail></Error>"
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.azure.core.http.rest.RestProxy.instantiateUnexpectedException(RestProxy.java:343)
	at com.azure.core.http.rest.RestProxy.lambda$ensureExpectedStatus$5(RestProxy.java:382)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:125)
	... 83 more
```

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report

- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by joshfree** [comment]
Thanks for filing this issue @sjw1066.  @ki1729 can you please follow up on this `customer-reported` issue in `Service Bus`? 

/cc @anuchandy

**Comment by ramya-rao-a** [comment]
This bug is already being tracked in #22657
So closing this issue as duplicate.

Please subscribe to #22657 for updates on this bug

**Comment by ramya-rao-a** [comment]
Thanks for your patience @sjw1066 

The fix for this issue was in the PR #25146

The November release for the azure-messaging-servicebus package was published last week with this fix.

Please use the latest version of the package i.e. 7.5.0 to get the fix.
